compute/openstack.py

```diff
@@ -128,6 +128,16 @@ def __init__(
             domain_name:    The authentication domain to be used.
             node_name:      The name of the node to be retrieved.
         """
+        # Store credentials for rehydration
+        self.username = username
+        self.password = password
+        self.auth_url = auth_url
+        self.auth_version = auth_version
+        self.tenant_name = tenant_name
+        self.tenant_domain_id = tenant_domain_id
+        self.service_region = service_region
+        self.domain_name = domain_name
+
```

Comment:
Can we avoid storing this information as the `driver` already has the necessary attributes and sessions.